src/tree/tree.ts

```diff
@@ -32,10 +33,27 @@ export class SchemaTreeListTree extends TreeListTree {
   public treeOptions: SchemaTreeOptions;
 
   public jsonSchemaTree!: JsonSchemaTree;
-  private _schemaToTreeMap!: WeakMap<SchemaNode, SchemaTreeListNode>;
-  private _treeToSchemaMap!: WeakMap<SchemaTreeListNode, SchemaNode>;
+  private _schemaToTreeMap!: WeakMap<SchemaNode, TreeListNode>;
+  private _treeToSchemaMap!: WeakMap<TreeListNode, SchemaNode>;
   private _flattenedSchemaNodes!: WeakSet<RegularNode>;
 
+  private _selectedIndex: number = 0;
+
+  @action
+  public updateSelectedTypeIndex = (treeListNode: TreeListNode, newIndex: number) => {
+    this._selectedIndex = newIndex;
+    this.replaceNode(treeListNode, this.buildTreeFragment((treeListNode.metadata as any).schemaNode.parent));
```

Comment:
Urgh. For a long time I've been willing to fix the signature of `metadata`.
It's such a PITA, particularly in Studio.